contribs/gnodev/pkg/dev/node.go

```diff
@@ -411,13 +368,76 @@ func (pm PkgsMap) Load(creator bft.Address, fee std.Fee, deposit std.Coins) ([]s
 	return txs, nil
 }
 
-func newNode(logger *slog.Logger, genesis gnoland.GnoGenesisState) (*node.Node, error) {
+func newNode(ctx context.Context, logger *slog.Logger, emitter emitter.Emitter, genesis gnoland.GnoGenesisState) (*node.Node, error) {
 	rootdir := gnoenv.RootDir()
 
+	// Setup node config
 	nodeConfig := gnoland.NewDefaultInMemoryNodeConfig(rootdir)
 	nodeConfig.SkipFailingGenesisTxs = true
 	nodeConfig.TMConfig.Consensus.SkipTimeoutCommit = false // avoid time drifting, see issue #1507
-
 	nodeConfig.Genesis.AppState = genesis
-	return gnoland.NewInMemoryNode(logger, nodeConfig)
+
+	// Generate a new node
+	var err error
+
+	// recoverError handles panics and converts them to errors.
+	recoverError := func() {
+		if r := recover(); r != nil {
+			panicErr, ok := r.(error)
+			if !ok {
+				panic(r) // Re-panic if not an error.
+			}
+
+			err = panicErr
+		}
+	}
+
+	createNode := func() *node.Node {
+		defer recoverError()
+
+		node, nodeErr := gnoland.NewInMemoryNode(logger, nodeConfig)
+		if nodeErr != nil {
+			err = fmt.Errorf("unable to create node: %w", nodeErr)
+			return nil
+		}
+
+		node.EventSwitch().AddListener("dev-emitter", func(evt tm2events.Event) {
+			switch data := evt.(type) {
+			case bft.EventTx:
+				resEvt := events.TxResult{
+					Height:   data.Result.Height,
+					Index:    data.Result.Index,
+					Response: data.Result.Response,
+				}
+
+				if err := amino.Unmarshal(data.Result.Tx, &resEvt.Tx); err != nil {
+					logger.Error("unable to unwarp tx result",
+						"error", err)
+				}
+
+				emitter.Emit(resEvt)
+			}
+		})
+
+		if startErr := node.Start(); startErr != nil {
+			err = fmt.Errorf("unable to start the node: %w", startErr)
+			return nil
+		}
+
+		return node
+	}
```

Comment:
Could we separate this as a private function instead?

```suggestion
defer recoverError()
n, err := createNode()
if err != nil {
    ...
}
```